### PR TITLE
Fixed getting router IP from itlwm interface

### LINKED ISSUE #104: [Bug] Connected WLAN's "Router" unknown in debug menu of HeliPort (but shown in SysPrefs)
Although this could be is an **itlwm.kext** issue, I am reporting it here as it involves UI as well. Please assign this accordingly. Thanks.

**Describe the bug**
On **HeliPort.app** and with the debug menu shown (via Option key on keyboard and clicking on the menubar icon) I can see the connected WLAN network and IP address obtained and other kext information, but "Router" remains "Unknown".

**To Reproduce**
1. With Option key on keyboard, click on HeliPort icon on menubar;
2. Verify the SSID is connected and data is displayed;
3. Check "Router" information displayed;
4. On both 2.4GHz and 5GHz Wi-Fi networks of the same router, it's shown as "Unknown";
5. Go to System Preferences and Network section;
6. Click on the Intel WLAN device: Router information is there.

**Expected behavior**
Not sure what's supposed to be shown in "Router" so I can guess the Gateway information that's 192.168.1.1 i.e. my Asus router's IP. This information is displayed just fine in System Preferences -> Network -> ITLWM device tab.

**Screenshots**
Please see below both HeliPort and System Preferences:

Router

Network

**Environment (please complete the following information):**
 - OS Version: Mojave 10.14.6
 - Software Version: Commits up to 2020-07-17

**Additional context**
I tested on both 2.4GHz and 5GHz of my Asus RT-AC58U router, same result. Thanks.

### PR DESCRIPTION
**Pull request by ErrorErrorError**
Hey!

This pull request fixes an issue with getting router IP address from itlwm interface, rather than getting it from the primary interface, which could've been on a different network or with a different router.

Should fix #104 

Thanks to @Goshin  for the workaround!

Ready for review!

### CODE REVIEW

File: HeliPort/NetworkManager.swift [hunk 1]
```diff
@@ -186,23 +186,16 @@ final class NetworkManager {
...
-            ?? SCDynamicStoreCopyValue(dynamicCreate, keyIPv6)
-
-        guard let interface = dictionary?[kSCDynamicStorePropNetPrimaryInterface] as? String, interface == bsd else {
-            Log.error("Could not find interface")
+        let routerCommand = ["netstat -rn | egrep -o default.*\(bsd)"]
+        guard let routerOutput = commandLine(args: routerCommand) else {
             return nil
         }
+        let ipv4Command = [#"echo "\#(routerOutput)" | egrep -o "\#(ipv4Pattern)""#]
+        let ipv4Output = commandLine(args: ipv4Command)
```
Comment (Goshin) [inline HeliPort/NetworkManager.swift:194]: You don't need to run a shell command to extract the address. Try extracting it with `NSRegularExpression` directly.

**Comment by Goshin** [comment]
The strict and complex regex patterns for ipv4 and ipv6 addresses are not necessary here, as the addresses of the result should be absolutely valid. To extract the addresses, you can simply use:

```
\s([0-9\.]+)\s          # for ipv4
\s([a-fA-F0-9:]+)\s     # for ipv6
```

File: HeliPort/NetworkManager.swift (hunk 1 above)
Comment (ErrorErrorError) [reply HeliPort/NetworkManager.swift:194]: Got it

**Comment by ErrorErrorError** [comment]
> The strict and complex regex patterns for ipv4 and ipv6 addresses are not necessary here, as the addresses of the result should be absolutely valid. To extract the addresses, you can simply use:
> 
> ```
> \s([a-fA-F0-9\.:]+)\s     # for ipv4 and ipv6
> ```

I tried using this regex to get IPv6 does not seem to work well on my end. Can you verify if it shows your IPv6 using this regex? Getting IPv4 using this works great however.

**Comment by Goshin** [comment]
@ErrorErrorError what is the netstat result of ipv6?

**Comment by ErrorErrorError** [comment]
I modified the regex since it had to do with the whitespace at the end, however it would also showed other values so I just added a minimum of `{4,}` so it wouldn't match those random characters.
```\s([a-fA-F0-9\.:]{4,})```

Screen Shot 2020-07-21 at 12 47 43 PM

**Comment by Goshin** [comment]
hmm, what is the result of `netstat -rn | egrep -o "default.*en0"`.

**Comment by ErrorErrorError** [comment]
> hmm, what is the result of `netstat -rn | egrep -o "default.*en0"`?

**Comment by ErrorErrorError** [comment]
Whoops, accidentally clicked closed, sorry! 




> hmm, what is the result of `netstat -rn | egrep -o "default.*en0"`?

I uploaded a picture that shows the steps I did to modify the regex, but the output I get when running `netstat -rn | egrep -o "default.*en0"` is 
```
~
➜ netstat -rn | egrep -o "default.*en0"                                            12:47:11

default            192.168.1.1        UGSc           en0
default                                 fe80::f6f5:e8ff:fe77:2ef1%en0   UGc             en0
```

**Comment by Goshin** [comment]
```
\s([a-fA-F0-9\.:]+)(\s|%)
```

Try using this one. You need to get the first matched group.

**Comment by ErrorErrorError** [comment]
> ```
> \s([a-fA-F0-9\.:]+)(\s|%)
> ```
> 
> Try using this one. You need to get the first matched group.

Okay, I am also getting the match group by using `NSRegularExpression.firstMatch` function. Would you still prefer the first match script version?

Also, I now ran with the regex you provided, but now the percentage is showing at the end. I can remove it via `replacingStrings` in NSString.
```
~
➜ netstat -rn | egrep -o "default.*en0" | egrep -o "\s([a-fA-F0-9\.:]+)(\s|%)"     12:47:38

 192.168.1.1
 fe80::f6f5:e8ff:fe77:2ef1%

```

**Comment by Goshin** [comment]
Obtaining the matched group will automatically get rid of the leading space and trailing `%`.

**Comment by ErrorErrorError** [comment]
> Obtaining the matched group will automatically get rid of the leading space and trailing `%`.

Yup I forgot to add the number in `range(at: Int)` and now it gets the first match without trailing `%` and whitespaces!

File: HeliPort/NetworkManager.swift [hunk 2]
```diff
@@ -287,3 +289,34 @@ final class NetworkManager {
         return ITL80211_SECURITY_UNKNOWN
     }
 }
+
+extension NetworkManager {
+    // from Goshin
+    private static let ipAddressRegex = #"\s([a-fA-F0-9\.:]+)(\s|%)"# // for ipv4 and ipv6
```
Comment (Goshin) [inline HeliPort/NetworkManager.swift:295]: Wouldn't it be better to declare it as a variable in `getRouterAddress`? as it is not a generic pattern for IP address, only useful for parsing `netstat -rn`.

Comment (ErrorErrorError) [reply HeliPort/NetworkManager.swift:295]: You're right. Let me remove that line and place it in `getRouterAddress`.

**Comment by Goshin** [comment]
Thanks to @ErrorErrorError for the great work.

**Pull request merged**